tests/conftest.py

```
@@ -0,0 +1,374 @@
+    Example:
+        @pytest.mark.parametrize("device_name", ALL_MOBILE_DEVICES)
+        def test_mobile(mobile_page, device_name):
+            mobile_page.goto(BASE_URL)
+            # ... test mobile-specific behavior
+    """
+    # Get device_name from parametrize
+    device_name = request.node.callspec.params.get("device_name")
+    if not device_name:
+        raise ValueError(
+            "mobile_page fixture requires 'device_name' parameter from @pytest.mark.parametrize"
+        )
```

Ruff (0.14.11)

Avoid specifying long messages outside the exception class

(TRY003)